internal/cmd/keys_test.go

```diff
@@ -25,6 +26,10 @@ func TestKeysAddCmd(t *testing.T) {
 		handler := func(resp http.ResponseWriter, req *http.Request) {
 			// the command does a lookup for machine ID
 			if requestMatches(req, http.MethodGet, "/v1/identities") {
+				if req.URL.Query().Get("name") != "my-machine" {
```

Comment:
Any ideas for how we can simplify these CLI test handlers? I could see a lot of code building up here. Not needed for this PR though.

Reply:
Ya, great question! I'm honestly not sure what to do with these yet. 

I'm guessing you are referring to the stub implementations of the API handlers? I wrote a few last week that seemed like they would be a challenge to maintain, so I have also been thinking we should look for other options there.

One option is to run a full `server.Server`, which removes the need to return canned data, but introduces the need to create the entities in the DB. I'm not sure how much extra time that will add to tests, but it's an option we can try out. That approach would also makes it much harder to write these tests, because we can't predict the IDs or times in the API responses.

If our API handlers accepted an interface for the "backend" (what is currently `access` and `data`) we could run a real API router, with a fake backend to inject the data. That might be less work than what we have here. That would be a much larger change, and probably not something we can easily do just for this purpose.

Another option would be to extract some of this, and make a more complete "fake" API. We could use an actual router that simulates our real API and put that in a more central place. That might allow us to remove some of the boilerplate path matching and JSON encoding, and allow these setup methods to only specify canned responses for specific inputs.

That last option seems the most viable at this point. Thoughts?

Reply:
(moved discussion to our engineering channel)